csharp/Managed/Pd.cs

```diff
@@ -0,0 +1,202 @@
+﻿using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using LibPDBinding.Native;
+
+namespace LibPDBinding.Managed
+{
+	/// <summary>
+	/// Class for a Pd instance. 
+	/// 
+	/// NB: Currently only one instance of Pd should be instanciated, but multi-process LibPd is on the way.
+	/// </summary>
+	public sealed class Pd : IDisposable
+	{
+		/// <summary>
+		/// Gets the number of input channels.
+		/// </summary>
+		public int Inputs { get; private set; }
+
+		/// <summary>
+		/// Gets the number of output channels.
+		/// </summary>
+		public int Outputs { get; private set; }
+
+		/// <summary>
+		/// Gets the sample rate.
+		/// </summary>
+		public int SampleRate { get; private set; }
+
+		/// <summary>
+		/// Gets the block size.
+		/// </summary>
+		public int BlockSize {
+			get { return PInvoke.blocksize (); }
+		}
+
+		/// <summary>
+		/// Returns [true] when audio computation is enabled, and [false] when audio computation is disabled.
+		/// </summary>
+		public bool IsComputing { get; private set; }
+
+		/// <summary>
+		/// Initializes a new instance of Pd
+		/// </summary>
+		/// <param name="inputChannels">Number of input channels.</param>
+		/// <param name="outputChannels">Number of output channels.</param>
+		/// <param name="sampleRate">Sample rate for project.</param>
+		public Pd (int inputChannels, int outputChannels, int sampleRate) : this (inputChannels, outputChannels, sampleRate, null)
+		{
+		}
+
+		/// <summary>
+		/// Initializes a new instance of Pd
+		/// </summary>
+		/// <param name="inputChannels">Number of input channels.</param>
+		/// <param name="outputChannels">Number of output channels.</param>
+		/// <param name="sampleRate">Sample rate for project.</param>
+		/// <param name="searchPaths">Paths for Pd to search for externals.</param>
+		public Pd (int inputChannels, int outputChannels, int sampleRate, IEnumerable<string> searchPaths)
```

Comment:
OK, so this is a design decision:
- should the consumer of this class be able to change the number of input and output channels later?
- should she be able to add search paths for externals later?